src/test.js

```diff
@@ -571,3 +571,27 @@ function skip( testName ) {
 
 	test.queue();
 }
+
+// Will be exposed as QUnit.only
+function only( testName, expected, callback, async ) {
+	var newTest;
+
+	if ( arguments.length === 2 ) {
+		callback = expected;
+		expected = null;
+	}
+
+	newTest = new Test({
+		testName: testName,
+		expected: expected,
+		async: async,
+		callback: callback
+	});
+
+	// First only wins, ignore subsequent onlys
+	if ( !QUnit.config.filter ) {
+		QUnit.config.queue.length = 0;
+		QUnit.config.filter = newTest.module.name + ": " + newTest.testName;
+		newTest.queue();
```

Comment:
it sounds weird, but this line should be placed after and outside the if block. 

If the filter is already set to run this `only` test (let's say I click on the rerun button on the reporter), I still need to call `newTest.queue()`.